terraform/gcp/modules/rekor/pubsub.tf

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2023 The Sigstore Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_project_iam_custom_role" "pubsub_publisher" {
+  project     = var.project_id
+  role_id     = "rekorPubsubPublisher"
+  title       = "Rekor PubSub Publisher"
+  description = "Able to inspect a topic and publish messages to it."
+  permissions = ["pubsub.topics.get", "pubsub.topics.publish"]
+}
+
+resource "google_pubsub_topic" "new_entry" {
```

Comment:
What do you think about also adding a flag to enable/disable this like https://github.com/sigstore/scaffolding/blob/main/terraform/gcp/modules/sigstore/sigstore.tf#L259C6-L259C6?